### PR TITLE
feature: make PROCESS_GAS immutable

### PR DESCRIPTION
**Pull request by prestwich**

### CODE REVIEW

**Review by emberian — APPROVED** [review]
lgtm

File: solidity/optics-core/contracts/Replica.sol [hunk 1]
```diff
@@ -86,7 +86,16 @@ contract Replica is Version0, Common {
     // ============ Constructor ============
 
     // solhint-disable-next-line no-empty-blocks
-    constructor(uint32 _localDomain) Common(_localDomain) {}
+    constructor(
+        uint32 _localDomain,
+        uint256 _processGas,
+        uint256 _reserveGas
+    ) Common(_localDomain) {
+        require(_processGas >= 850_000, "!process gas");
```
Comment (anna-carroll) [inline solidity/optics-core/contracts/Replica.sol:94]: why underscore here?

**Review by anna-carroll — APPROVED** [review]
LGTM

File: solidity/optics-core/contracts/Replica.sol (hunk 1 above)
Comment (anna-carroll) [reply solidity/optics-core/contracts/Replica.sol:94]: seems like this is acceptable syntax for spacing number literals

Comment (prestwich) [reply solidity/optics-core/contracts/Replica.sol:94]: yes, the syntax was introduce to make numeric literals more legible

**Pull request merged**